src/extractor/guidance/turn_handler.cpp

```diff
@@ -191,15 +209,12 @@ bool TurnHandler::isObviousOfTwo(const EdgeID via_edge,
 
 bool TurnHandler::hasObvious(const EdgeID &via_edge, const Fork &fork) const
 {
-    for (auto road = fork.right; road < fork.left; ++road)
-    {
-        if (isObviousOfTwo(via_edge, *road, *(road + 1)) ||
-            isObviousOfTwo(via_edge, *(road + 1), *road))
-        {
-            return true;
-        }
-    }
-    return false;
+    auto obvious_road = std::adjacent_find(
+        fork.begin, fork.end, [&](const ConnectedRoad &a, const ConnectedRoad &b) {
```

Comment:
Capturing (heh) from chat: you have to capture `this` for being able to call a member function.

All member function calls `memfn` are implicitely `this->memfn()`.